### ISSUE TITLE
Regenerate road segments when material updated

### ISSUE THREAD

**Issue by TheDuckCow** [TheDuckCow/godot-road-generator#79]
If the user assigns a new material to the RoadNetwork, it should add an updated material. Right now, the user would have to trigger a full reload of segments manually (most easily right now by toggling the auto refresh button on and off).

Ideally, it can do this in a lightweight way - iterating through existing meshes and updating their material slot, vs completely regenerating geometry.

**Comment by bdog2112** [comment]
**Please describe steps to recreate problem.** I changed the road texture several times and it always updated instantly.

Steps that I followed:
- Duplicate road_texture.png, invert colors, and rename new image to "road_texture_inverted.png"
- Select RoadNetwork in Scene dock
- Select "Material Resource" in Inspector dock
- Expand "Albedo" to expose "Texture"
- Drag/Drop road_texture_inverted.png or road_texture.png from FileSystem dock into "Texture" slot in Inspector dock
- Outcome: New image is displayed on all Segments in 3D View

Also tried:
- Duplicate road_texture.material and rename to road_texture2.material
- Change albedo texture to road_texture_inverted.png in new material
- Select RoadNetwork in Scene dock
- Drag/Drop road_texture.material or road_texture2.material from FileSystem dock into "Material Resource" slot in Inspector dock
- Outcome: New material is displayed on all Segments in 3D View

**Comment by TheDuckCow** [comment]
Hm, I've retried this myself now and can no longer replicate it. It must have been resolved with some of the other updates we made around how and when refreshing works. I checked out version 0.2.0 for comparison, it was definitely an issue still then, and now is not. Even if I drag and drop a new material onto the network slot now, it immediately updates the road. Closing this issue, apologies for that!
